orttraining/orttraining/core/optimizer/graph_transformer_utils.cc

```diff
@@ -80,6 +81,8 @@ std::vector<std::unique_ptr<GraphTransformer>> GeneratePreTrainingTransformers(
       // Remove duplicate nodes. Must be applied before any recompute transformations.
       transformers.emplace_back(onnxruntime::make_unique<CommonSubexpressionEliminationApplyOnce>(compatible_eps));
 
+      transformers.emplace_back(onnxruntime::make_unique<NotWhereFusion>(compatible_eps));
```

Comment:
I think this optimization is also harmless to add in the InferenceSession's graph transformer list. 
Please check with "ONNX Runtime Core" alias.

Reply:
I have checked with them, adding